libmscore/paste.cpp

```diff
@@ -981,7 +980,14 @@ void Score::cmdPaste(const QMimeData* ms, MuseScoreView* view, Fraction scale)
                               ddata.dropElement = nel;
                               ddata.duration    = duration;
                               if (target->acceptDrop(ddata)) {
+                                    // dropping an element of the same type is likely to replace it
+                                    // thus invaldiating the selection
+                                    ElementType targetType = target->type();
+                                    if (targetType == type)
+                                          deselect(target);
                                     target->drop(ddata);
+                                    if (targetType == type)
+                                          select(nel);
                                     if (_selection.element())
```

Comment:
The above does the same check, `if (targetType == type)`, twice every time. Wouldn't
`if (targetType == type) {
deselect(target);
target->drop(ddata);
select(nel);
}
else
target->drop(ddata);
`, with only one check, be the recommended speed optimization?

Reply:
Yes.  But since this happens only on execution of a specific command, not on every layout, and we're talking about a simple bool comparison, I feel it is better to reduce code duplication than to optimize for speed.  A judgment call to be sure.

Reply:
FWIW, my original code turned out not to work for dropping clefs onto clefs etc, so I needed to revisit this, and ended up taking your advice - see #5098